### PR TITLE
[Reviewer: Andy] Interface for MemcachedConfig code

### PR DESCRIPTION
**Pull request by AlexHockey**
This change defines the interface to the memcached config reading code I am writing. Please could you take a look at it and let me know if this works for you? 

Note that I have defined separate classed for `MemcachedConfigReader` (the interface) and `MemcachedConfigFileReader` (the implementation). I think this is cleaner and encourages looser coupling between the classes that use the config. For the same reason, I have also taken your suggestion of changing `MemcachedStoreView::update` to take a config object.

### CODE REVIEW

**Comment by AlexHockey** [comment]
BTW, please don't merge this :-)

File: include/memcached_config.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+  std::vector<std::string> new_servers;
+
+  /// The expiry time (in seconds) of tombstone records that are written when
+  /// data is deleted from memcached.
+  ///
+  /// 0 means that tombstones are not used. Deleted data is simply removed.
+  int tombstone_lifetime;
+
+  /// Default constructor.
+  MemcachedConfig();
```
Comment (bossmc) [inline include/memcached_config.h]: I don't think you need a default constructor unless the default value for one of the variables is not that variable's default value.  In this case I assume the defaults are empty lists and 0 which would be the automatic defaults.

**Comment by bossmc** [comment]
Looks good.  Happy for you to go ahead with implementation.

File: include/memcached_config.h (hunk 1 above)
Comment (AlexHockey) [inline include/memcached_config.h]: Not quite. The default is currently 0 but will be changing to a non-zero value in the next Project Clearwater release. (Also, I *think* that POD types are never default-initialized anyway, but I could be wrong about that).

Comment (bossmc) [inline include/memcached_config.h]: Ok, sounds reasonable.

**Comment by AlexHockey** [comment]
@bossmc, I've added the implementation. Please could you review this as well?

File: include/memcachedstore.h [hunk 2]
```diff
@@ -49,30 +49,35 @@ extern "C" {
...
+//
+// This file contains the definition of a memcached-based store. This is split
+// across two classes:
+// -  BaseMemcachedStore which contains the majority of the code.
+// -  MemcachedStore. This is the class that users of the store should
+//    create.
+//
+// MemcacheStore is separate class as this allows different constructors to be
+// defined which all delegate to the BaseMemcachedStore class. This is required
+// as BaseMemcachedStore contains constant memebers, and our version of GCC
```
Comment (bossmc) [inline include/memcachedstore.h]: sp: "memebers"

File: include/memcachedstore.h [hunk 3]
```diff
@@ -49,30 +49,35 @@ extern "C" {
...
+// does not support delegating to constructors in the same class.
```
Comment (bossmc) [inline include/memcachedstore.h]: You could use the old `init` function trick to commonalize construction, but what you've done is fine (shame we don't have proper C++11 support).

**Comment by bossmc** [comment]
Looks fine.  One spelling mistake.  I'll fix it when pull into my integration branch unless you want to rush this into trunk?

File: include/memcachedstore.h (hunk 3 above)
Comment (AlexHockey) [inline include/memcachedstore.h]: Actually I can't. the old `BaseMemcachedStore` has some constant members that have to be initialized in the initializer list.

**Comment by AlexHockey** [comment]
Cheers. I have no urgent desire to get this into trunk on it's own (and there is no point both of us updating the submodule refs), so I'm closing the PR.

**Pull request closed without merging**